Python/optimizer_bytecodes.c

```diff
@@ -446,7 +446,26 @@ dummy_func(void) {
     }
 
     op(_COMPARE_OP_INT, (left, right -- res)) {
-        res = sym_new_type(ctx, &PyBool_Type);
+        if (sym_is_const(ctx, left) && sym_is_const(ctx, right) &&
+        sym_matches_type(left, &PyLong_Type) && sym_matches_type(right, &PyLong_Type))
+        {
```

Comment:
I know the other instructions do this, but I don't think it's necessary. The types of these symbols should have been successfully narrowed by this point.

Either way, the asserts below will catch if we made a mistake.

```suggestion
        if (sym_is_const(ctx, left) && sym_is_const(ctx, right)) {
```